### PR TITLE
Add possibility to hide function bar in MainPanel

### PR DESCRIPTION
**Pull request by cgzones**
Also fix the selection-cursor currently getting hidden by the FunctionBar when scrolling down.

### CODE REVIEW

**Comment by BenBE** [comment]
Typo in commit message:
hight -> height

File: Panel.c [hunk 1]
```diff
@@ -336,10 +339,14 @@ void Panel_draw(Panel* this, bool force_redraw, bool focus, bool highlightSelect
...
+   const int size = Vector_size(this->items);
 
-   #define CLAMP_INDEX(var, delta, min, max) \
-      CLAMP((var) + (delta), (min), MAXIMUM(0, (max)))
+   #define PANEL_SCROLL(amount)                                                                                                                                      \
+   do {                                                                                                                                                              \
+      this->selected += (amount);                                                                                                                                    \
+         this->scrollV = CLAMP(this->scrollV + (amount), 0, MAXIMUM(0, (size - (this->h + this->hightModifier) - (RichString_sizeVal(this->header) > 0 ? 1 : 0))));  \
+         this->needsRedraw = true;                                                                                                                                   \
+      } while (0)
```
Comment (BenBE) [inline Panel.c:349]: indentation

**Review by BenBE — CHANGES_REQUESTED** [review]
Overall typo: `hight` -> `height`

**Comment by fasterit** [comment]
Please make "Esc" toggle the function bar (and probably also the selection cursor bar).
Otherwise this is odd from a usability perspective.

People will very soon ask for a setting to hide the Function bar permanently if this PR gets merged, cf.  #168.
So you should probably implement that at the same time.

I'd personally prefer a simple setting to this Esc logic which may annoy the people that just want to get rid of the selection bar but keep those unsatisfied that want to always get rid of the Function bar.

**Comment by BenBE** [comment]
I'd suggest not joining the ESC hiding of the cursor and the overall function bar hiding. IMO these are different things and should be kept separate. Based on demand in #168 I'd even argue that warrants a config option for the function bar hiding; though the ESC behaviour should be kept unimpeded by that setting.

**Review by BenBE — APPROVED** [review]
LGTM.

**Pull request merged**